File: code/ch09/Person.java
public class Person {
    private String name;
    private int age;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    void hello(Person p) {
        System.out.println("Hello, " + p.getName() + "!");
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public static void main(String[] args) {
        Person p = new Person("Chen", 5);
        p.hello(p);
        System.out.println("(He is " + p.getAge() + " years old.)");
    }
}

File: code/ch09/Person.cc
#include <iostream>
#include <string>

using namespace std;

class Person {
private:
  string name;
  int age;

public:
  Person(string, int);
  void hello(Person person);
  string getName() { return name; };
  void setName(string name) { this->name = name; }
  int getAge() { return age; };
  void setAge(int age) { this->age = age; }
};

Person::Person(string name, int age) {
  this->name = name;
  this->age = age;
}

void Person::hello(Person person) {
  cout << "Hello, " << person.getName() << "!" << endl;
}

int main() {
  Person p("Chen", 5);
  p.hello(p);
  cout << "(He is " << p.getAge() << " years old.)" << endl;
  delete &p;
  return 0;
}
